crates/accelerate/src/synthesis/permutation/mod.rs

```diff
@@ -82,11 +82,44 @@ fn _synth_permutation_acg(py: Python, pattern: PyArrayLike1<i64>) -> PyResult<Ci
     )
 }
 
+/// Synthesize a permutation circuit for a linear nearest-neighbor
+///     architecture using the Kutin, Moulton, Smithline method.
+#[pyfunction]
+#[pyo3(signature = (pattern))]
+pub fn _synth_permutation_depth_lnn_kms(
+    py: Python,
+    pattern: PyArrayLike1<i64>,
```

Comment:
In most of our synthesis code, we are using `PyReadonly1` instead of `PyArrayLike1`, however the current form is consistent with other methods in `synthesis/permutation/mod.rs`. So I am suggesting to leave this as is, but possibly refactor this file in a follow-up.